mm2src/coins/utxo/utxo_sql_block_header_storage.rs

```diff
@@ -293,6 +293,181 @@ impl BlockHeaderStorageOps for SqliteBlockHeadersStorage {
     }
 }
 
+#[cfg(test)]
+#[derive(Debug, Clone)]
+pub struct BlockHeaderInMemoryStorage {
+    pub ticker: String,
+    pub conn: Arc<Mutex<Connection>>,
+}
```

Comment:
This can be simplified significantly since `BlockHeaderInMemoryStorage` is used in tests only:
```rust
pub struct BlockHeaderInMemoryStorage {
    pub ticker: String,
    // The block headers should be ordered to be able to return the last block header.
    pub block_headers: BTreeMap<u64, BlockHeader>,
    // This can be used on [`BlockHeaderStorageOps::get_block_height_by_hash`].
    pub block_headers_by_hash: HashMap<H256, BlockHeader>,
}
```

Please also note that `get_last_block_height` should be configurable from a test. For example, at the beginning of a test you can just upload one block header with a specified height. Also there could be a field like `BlockHeaderInMemoryStorage::last_block_height`.

Reply:
noted for next PR. THANKS